atomic_reactor/__init__.py

```diff
@@ -32,6 +35,28 @@ def format(self, record):
         return super(ArchFormatter, self).format(record)
 
 
+class EncodedStream(object):
+    def __init__(self, fileno, encoding):
+        self.binarystream = fdopen(dup(fileno), 'wb')
```

Comment:
Unfortunately I have no suggestions, only questions :(

I ask because the encoding is used when writing to the file, not when opening the file.
So in theory, the same file descriptor should work.

In any case, it would be nice to have a warning comment in there explaining the intent.
This will, at least attempt to, prevent someone, like me, breaking things in the future.